src/components/Icon/Icon.tsx

```diff
@@ -5,68 +5,149 @@ import svg4everybody from 'svg4everybody';
 import styles from './Icon.module.css';
 import icons from '../../icons/spritemap/spritemap.svg';
 
-export interface Props {
+interface IconPropsBase {
   /**
    * CSS class names that can be appended to the component.
    */
   className?: string;
   /**
-   * Declares whether the icon is focusable or not for accessibility
+   * Either a fragment of svg elements (g, path, circle, etc.), or a single
+   * svg element. Useful for creating specific icon components.
+   *
+   * @example
+   * function CircleIcon(props: IconProps) {
+   *   return (
+   *     <Icon {...props}>
+   *       <circle cx="50" cy="50" r="50" />
+   *     </Icon>
+   *   )
+   * }
    */
-  focusable?: boolean;
+  children?: React.ReactNode;
+  /**
+   * The SVG Color, expects a valid css color (hex, rgb, etc.).
+   * Recommendation: if `currentColor` isn't sufficient, use color tokens from
+   * @chanzuckerberg/eds/lib/tokens/colors.ts
+   */
+  color?: string;
```

Comment:
I looked at places where `Icon` is being used and each time its fill color is being set by adding a `__icon` subclass to the parent component, and that fill color is overriding any value pass to the `color` prop of `Icon` (and in keeping with that pattern, I wasn't able to find any uses of `Icon` that actually did pass a value to `color`. But using this pattern of modifying one component from within another component via a subclass is icky, not SOLID, etc.

If we take this on and refactor the way props are sent to Icons, it would help to include something in CODE_GUIDELINES to say "do not use classes on one component to alter the behavior of a different component", which would mean no more adding classnames with other component names in them like "__icon"

Reply:
yep, this is a prop from the `main` branch. and agree that I like this more than the `__icon` class pattern -- will file a ticket to do this refactor and update the guidelines